kola/tests/systemd/sysext.go

```diff
@@ -50,3 +66,17 @@ func checkSysextSimple(c cluster.TestCluster) {
 	// Second check after reloading the extensions (e.g., to add/remove/update them)
 	checkHelper(c)
 }
+
+func checkSysextCustomDocker(c cluster.TestCluster) {
+	cmdNotWorking := `if docker run --rm ghcr.io/kinvolk/busybox true; then exit 1; fi`
+	cmdWorking := `docker run --rm ghcr.io/kinvolk/busybox echo Hello World`
+	// First assert that Docker doesn't work because Torcx is disabled
+	_ = c.MustSSH(c.Machines()[0], cmdNotWorking)
+	// We build a custom sysext image locally because we don't host them somewhere yet (TODO: use "main" branch)
+	_ = c.MustSSH(c.Machines()[0], `git clone https://github.com/flatcar/sysext-bakery.git && git -C sysext-bakery checkout kai/containerd`)
+	// Flatcar has no mksquashfs and btrfs is missing a bugfix but at least ext4 works
+	_ = c.MustSSH(c.Machines()[0], `ONLY_DOCKER=1 FORMAT=ext4 sysext-bakery/create_docker_sysext.sh 20.10.21 docker && ONLY_CONTAINERD=1 FORMAT=ext4 sysext-bakery/create_docker_sysext.sh 20.10.21 containerd && sudo mv docker.raw containerd.raw /etc/extensions/`)
```

Comment:
How about avoiding the hardcoding of the version in favor of something like `curl -s https://api.github.com/repos/moby/moby/releases/latest | jq --raw-output '.tag_name' | tail --bytes +2`?

Reply:
The idea of using a particular kola version referenced by `scripts` was to be reproducible. I would also use a git commit ID in the git check out for the same reason.

Reply:
I meant the docker version that is passed to the `create_docker_sysext.sh` script, not kola. Or maybe we could take the version from the docker ebuild in coreos-overlay, but not sure how to do it here.

Reply:
Yes, if the docker version is dynamic we lose the benefit of pinning kola tests for reproducibility.

Reply:
Alright, let's get this in for now. But would be great to eventually use a docker version that matches the one we package for tested flatcar version. That should be still relatively reproducible. Maybe something like:

```bash
docker_version=$(curl -fsSL https://bincache.flatcar-linux.net/images/${ARCH}/${VERSION}/flatcar_production_image_packages.txt | grep 'app-emulation/docker-[0-9]' | cut -d- -f3 | cut -d: -f1)
```

Otherwise we will need another github action that bumps the docker version from time to time to use something recent.

Reply:
Good idea to bind it to the release.
While it makes sense to test recent versions it's also meaningful to test old versions because when someone wants to use a custom Docker version (and therefore, uses sysext), it's likely that they don't update it and stick to the old version.

Reply:
For binding it to the Docker version inside the image we could get the version from the license json file under `/usr/share/flatcar/`.

Reply:
I'll update the test to use both and then we can explain why we want to also use an old version.